kale/embed/uncertainty_fitting.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import os
+from typing import List, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+from yacs.config import CfgNode
+
+from kale.interpret.uncertainty_quantiles import quantile_binning_and_est_errors
+from kale.loaddata.tabular_access import load_csv_columns
+from kale.predict.uncertainty_binning import quantile_binning_predictions
+from kale.prepdata.tabular_transform import apply_confidence_inversion
+
+
+def fit_and_predict(
+    landmark: int,
+    uncertainty_error_pairs: List[List],
+    ue_pairs_val: str,
+    ue_pairs_test: str,
+    num_bins: int,
+    config: CfgNode,
+    groundtruth_test_errors: bool,
+    save_folder: Optional[str] = None,
+) -> Tuple[pd.DataFrame, pd.DataFrame, pd.DataFrame]:
+    """
+    Loads (validation, testing data) pairs of (uncertainty, error) pairs and for each fold: uses the validation
+    set to generate quantile thresholds, estimate error bounds and bin the test data accordingly. Saves
+    predicted bins and error bounds to a csv.
+
+    Args:
+        landmark (int): Which landmark to perform uncertainty estimation on.
+        uncertainty_error_pairs (list[list]): List of lists describing the different uncertainty combinations to test.
+        ue_pairs_val (str): Path to validation pairs (uncertainty, error) data.
+        ue_pairs_test (str): Path to test pairs (uncertainty, error) data.
+        num_bins (int): Number of bins for quantile binning.
+        config (CfgNode): Configuration object with hyperparameters and other settings.
```

Comment:
Making the hyperparameters explicit would be more readable.

Reply:
what do you mean explicit? default values?

Reply:
for future, this mean removing CfgNode and explicitly passing used params

Reply:
Have you captured this in the todo list in discussion? Otherwise, how are we going to keep track?